### PR TITLE
feat: initialize and open source email plugin code

### PR DESCRIPTION
**Pull request by kaymckay**
Adding Initial email code to external repo

### CODE REVIEW

File: cmd/vela-email/plugin.go
Comment (davidvader) [inline cmd/vela-email/plugin.go:98]: what do these comments mean?
```
		// Don't check email's errors
```

**Review by davidvader — APPROVED** [review]
LGTM, nice work

File: cmd/vela-email/plugin.go
Comment (kaymckay) [reply cmd/vela-email/plugin.go:98]: They can be removed, it means just return the error that the email code returns, don't double check that they returned the right error and don't create a new one

File: cmd/vela-email/plugin.go
Comment (JordanSussman) [inline cmd/vela-email/plugin.go:141]: Can these be set as optional values instead? I have a use-case where I want to send emails through a smtp server that does require or support authentication.

Comment (kaymckay) [reply cmd/vela-email/plugin.go:141]: Which sendtype would you need to use? each of our sendtypes require authentication

Comment (JordanSussman) [reply cmd/vela-email/plugin.go:141]: I'm not familiar with the github.com/jordan-wright/email package you are using, but the code comments make it seem like SMTP auth is optional for some of the functions - https://github.com/jordan-wright/email/blob/master/email.go#L507. I have also used gopkg.in/mail.v2 in the past to send emails without authentication by doing something like:

```go
m := mail.NewMessage()
m.SetHeader("From", cfg.EmailFrom)
m.SetHeader("To", email)
m.SetHeader("Subject", cfg.EmailSubject)
m.SetBody("text/html", body)

d := mail.Dialer{Host: cfg.SMTPRelay, Port: 25}
if err := d.DialAndSend(m); err != nil {
  return err
}
```

**Review by plyr4 — DISMISSED** [review]
LGTM, appreciate the optional auth changes.

**Comment by GregoryDosh** [comment]
Correct me if I'm wrong, but I seem to recall someone like @wass3r mentioning that the way that the changelogs are done historically was causing some clashing (& thrashing with overwrites) with using built-in GitHub changelog creation. Do we need to make updates to the workflows to remove that old changelog plugin and functionality in favor of GitHub's built-in?

File: cmd/vela-email/plugin_test.go
Comment (GregoryDosh) [reply cmd/vela-email/plugin_test.go:79]: Adding on to this, there are some RFCs around that outline some ways of referencing technical things in documentation and examples. For emails and top level domains and the like, there is this RFC https://datatracker.ietf.org/doc/html/rfc2606 which has couple of options to pick from that should work for tests and docs. Specifically example.com is great because if you navigate to that website it even explicitly says you can use it without permission.

File: cmd/vela-email/main.go
Comment (GregoryDosh) [inline cmd/vela-email/main.go:280]: This feels a little awkward to me. We could probably remove the `Env` field and anywhere the plugin was using that field we can use the `Environment()` method instead. I don't think there are many (just 1?) reference to this so it's small but it'd make things a little simpler.

File: cmd/vela-email/plugin.go
Comment (GregoryDosh) [inline cmd/vela-email/plugin.go:258]: I think this length check4 code isn't needed. If you range over a length 0 slice it shouldn't have anything in it. So you should be able to remove this code without impacting any of the function or logic.

**Review by GregoryDosh — APPROVED** [review]
Great work! Thanks for taking the time and effort to make this available for the community.

There were some small nits I included above but they're not show stoppers. I'll approve for now and if you need a re-approval later lemme know.

**Review by GregoryDosh — APPROVED** [review]

**Review by plyr4 — APPROVED** [review]

**Pull request merged**
